x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/common/io/SqlStreamInput.java

```diff
@@ -25,11 +25,11 @@
  */
 public class SqlStreamInput extends NamedWriteableAwareStreamInput {
 
-    public static SqlStreamInput fromString(String base64encoded, NamedWriteableRegistry namedWriteableRegistry, Version version)
+    public static SqlStreamInput fromString(String base64encoded, NamedWriteableRegistry namedWriteableRegistry, TransportVersion version)
```

Comment:
@Luegg Does this change make sense? What does `version` actually represent?

Reply:
The version here is the version of the node that created the cursor. Currently it's only used to ensure that we're using only cursors that have been created with the same version of ES. I guess if there are cases where the transport version can differ from the current node version then the transport version would not be entirely accurate here.

The cursor works as follows:

* User issues a query using the `_sql` endpoint. The response includes a cursor if the result set has not yet been exhausted. The cursor includes all instructions to fetch the next `n` rows of the result set (serialized with `Writeable`s and encoded as a base64 string).
* To fetch the next page, the user has to set the cursor in the next call to the `_sql` endpoint. The response might once again include an updated cursor to fetch the next page.

During upgrades, we do not yet support cursor compatibility.

All things considered I'm not quite sure what would be the best option here. I'd expect though that a change in TransportVersion during an SQL "transaction" without a change in node version would be rather untypical.

Reply:
Looks like this is best left as `Version` then

Reply:
I guess we still need to follow up to work around the call to the deprecated `setVersion` in https://github.com/elastic/elasticsearch/pull/93384/files#diff-179989ecaed3a7fbe1140bd37c5940581f0e7a80c903a8a96c52f7c17fbc5fdcL46 ? Or can we leave that as is for now?